### PR TITLE
Yasm Conan 2.0 compatibility

### PR DESCRIPTION
**Pull request by System-Arch**
Specify library name and version:  **yasm/1.3.0**

Cleaned up a few things and got tests to pass with Conan 2.0

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — APPROVED** [review]

File: recipes/yasm/all/conanfile.py [hunk 1]
```diff
@@ -4,14 +4,15 @@
 from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir
 from conan.tools.gnu import Autotools, AutotoolsToolchain
 from conan.tools.layout import basic_layout
-from conan.tools.microsoft import is_msvc, unix_path
+from conan.tools.microsoft import is_msvc
 import os
 
-required_conan_version = ">=1.52.0"
+required_conan_version = ">=1.56.0"
```
Comment (SpaceIm) [inline recipes/yasm/all/conanfile.py:10]:
```suggestion
required_conan_version = ">=1.54.0"
```

File: recipes/yasm/all/test_package/conanfile.py [hunk 2]
```diff
@@ -7,9 +7,9 @@ class TestPackageConan(ConanFile):
     generators = "VirtualRunEnv"
```
Comment (SpaceIm) [inline recipes/yasm/all/test_package/conanfile.py:7]:
```suggestion
    generators = "VirtualBuildEnv"
```

File: recipes/yasm/all/test_package/conanfile.py [hunk 3]
```diff
@@ -7,9 +7,9 @@ class TestPackageConan(ConanFile):
...
     test_type = "explicit"
 
-    def requirements(self):
-        self.requires(self.tested_reference_str)
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
         if can_run(self):
```
Comment (SpaceIm) [inline recipes/yasm/all/test_package/conanfile.py:14]:
```suggestion
```

File: recipes/yasm/all/test_package/conanfile.py [hunk 4]
```diff
@@ -7,9 +7,9 @@ class TestPackageConan(ConanFile):
...
-            self.run("yasm --help", env="conanrun")
+            self.run("yasm --help")
```
Comment (SpaceIm) [inline recipes/yasm/all/test_package/conanfile.py:15]:
```suggestion
        self.run("yasm --help")
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Comment by SpaceIm** [comment]
And `from conan.tools.build import can_run` can be removed now from conanfile of test package

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**